xc/xc7/yosys/synth.tcl

```diff
@@ -33,7 +33,38 @@ if { [info exists ::env(INPUT_XDC_FILE)] && $::env(INPUT_XDC_FILE) != "" } {
 write_verilog $::env(OUT_SYNTH_V).premap.v
 
 # Map Xilinx tech library to 7-series VPR tech library.
-read_verilog -lib $::env(TECHMAP_PATH)/cells_sim.v
+read_verilog -specify -lib $::env(TECHMAP_PATH)/cells_sim.v
+
+# Convert congested CARRY4 outputs to LUTs.
+techmap -map  $::env(TECHMAP_PATH)/carry_map.v
+write_json $::env(OUT_JSON).carry_fixup.json
+exec $::env(PYTHON3) $::env(TECHMAP_PATH)/fix_carry.py < $::env(OUT_JSON).carry_fixup.json > $::env(OUT_JSON).carry_fixup_out.json
+design -push
+read_json $::env(OUT_JSON).carry_fixup_out.json
+
+techmap -map  $::env(TECHMAP_PATH)/clean_carry_map.v
+
+# Re-read baseline libraries
+read_verilog -lib -specify -D_EXPLICIT_CARRY +/xilinx/cells_sim.v
+read_verilog -lib +/xilinx/cells_xtra.v
+read_verilog -specify -lib $::env(TECHMAP_PATH)/cells_sim.v
+if { $::env(USE_ROI) != "TRUE" } {
+    read_verilog -lib $::env(TECHMAP_PATH)/iobs.v
+}
+
+# Re-run optimization flow to absorb carry modifications
+hierarchy -check -auto-top
+
+write_ilang $::env(OUT_JSON).pre_abc9.ilang
+if { $::env(USE_ROI) == "TRUE" } {
+    synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp -noiopad -nowidelut -run map_ffs:check
+} else {
+    synth_xilinx -flatten -abc9 -nosrl -noclkbuf -nodsp -iopad -nowidelut -run map_ffs:check
```

Comment:
How does this re-synthesis step impact on run-time?

Reply:
I haven't measured it yet.